server/src/main/java/io/crate/execution/dml/ArrayIndexer.java

```diff
@@ -100,23 +107,26 @@ public static <T> ValueIndexer<T> of(Reference arrayRef, Function<ColumnIdent, R
     public static final String ARRAY_LENGTH_FIELD_PREFIX = "_array_length_";
 
     protected final ValueIndexer<T> innerIndexer;
-    private final String arrayLengthFieldName;
-    private final Reference reference;
+    protected final Streamer<List<T>> innerStreamer;
+    protected final String arrayLengthFieldName;
+    protected final Reference reference;
 
-    private ArrayIndexer(ValueIndexer<T> innerIndexer, Function<ColumnIdent, Reference> getRef, Reference reference) {
+    protected ArrayIndexer(ValueIndexer<T> innerIndexer, Function<ColumnIdent, Reference> getRef, Reference reference) {
         this.innerIndexer = innerIndexer;
-        this.arrayLengthFieldName = toArrayLengthFieldName(reference, getRef);
+        this.innerStreamer = (Streamer<List<T>>) reference.valueType().streamer();
```

Comment:
the term  `innerStreamer` confused me at first as we usually refer to the inner element of an array with `inner`.
Maybe lets call it for what it is used?
```suggestion
        this.bytesConverter = (Streamer<List<T>>) reference.valueType().streamer();
```